src/access/AccessControl.h

```diff
@@ -629,13 +632,15 @@ class AccessControl
 
     /**
      * Check whether access (by a subject descriptor, to a request path,
-     * requiring a privilege) should be allowed or denied.
+     * requiring a privilege) should be allowed or denied. The IM action
+     * is provided for context.
      *
      * @retval #CHIP_ERROR_ACCESS_DENIED if denied.
      * @retval other errors should also be treated as denied.
      * @retval #CHIP_NO_ERROR if allowed.
      */
-    CHIP_ERROR Check(const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath, Privilege requestPrivilege);
+    CHIP_ERROR Check(const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath, Privilege requestPrivilege,
```

Comment:
please fix the order to keep request last. It seems the intent was request data (descriptor, path) to be first, then the request last.

Also this starts growing into having too many arguments. We may decide to have a builder pattern starting now, something like `Check(Request, Privilege)` where Request is a builder with SetDescriptor, SetPath, SetMsgType).

Reply:
done.